src/object.d

```diff
@@ -3190,11 +3199,24 @@ void __ctfeWriteln(T...)(auto ref T values) { __ctfeWrite(values, "\n"); }
  * Create RTInfo for type T
  */
 
+template RTInfoImpl(size_t[] pointers)
+{
+    immutable size_t[pointers.length] data = pointers[];
+    immutable RTInfoImpl = data.ptr;
```

Comment:
As said for the original PR, we should spent a bit of time to save space and indirection, b/c having the metadata in cache will be essential for decent performance.
I'd suggest to convert the `size_t[]` first to a `ubyte[]` array (while leaving out all 0 values at the end), then use [Variable-length quantity](https://en.wikipedia.org/wiki/Variable-length_quantity) variable length encoding to store the array length in front of the metadata, and store it as a `ubyte*`.
This is a fairly simple optimization that will save quite some space (empiric numbers welcome) and removes one indirection.
For any struct below 64-byte this would only require `8+1+1=10 bytes` instead of `8+16+8=32 bytes`. Truncating zeros at the end will save on structs with big buffers as last field.
We should also use null pointers for data that doesn't require scanning, and could use 0 length as an escape magic preceeding more complex encodings, e.g. run length encoding for structs with big nogc holes.